### PR TITLE
fix: auto destroy CLOSE_WAIT socket

### PR DESCRIPTION
**Pull request by gxcsoccer**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

…T status

image

### CODE REVIEW

File: lib/base.js [hunk 1]
```diff
@@ -369,17 +382,21 @@ class TCPBase extends Base {
...
-        while (remaining);
+        } while (remaining);
       } catch (err) {
         this.close(err);
       }
     });
 
+    // receive `end` event that means the other end of the socket sends a FIN packet
+    socket.once('end', () => {
+      this.logger.info('the connection: %s is closed by other side', this[addressKey]);
```
Comment (fengmk2) [inline lib/base.js]: `'[tcp-base] the connection: %s is closed by other side'`

**Comment by fengmk2** [comment]
后面是不是内部的 tcp-base 就可以去掉了？都用这个开源的版本。

File: lib/base.js (hunk 1 above)
Comment (gxcsoccer) [reply lib/base.js]: ok

**Comment by gxcsoccer** [comment]
> 后面是不是内部的 tcp-base 就可以去掉了？都用这个开源的版本。

是的，得统一来次清理，记个 issue

---

我看了下用的地方没几个，这个合并以后我就去替换

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
1.0.2

**Pull request merged**